ProcessLib/Parameter/CoordinateSystem.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CoordinateSystem.h"
+
+#include <typeinfo>
+
+#include "Parameter.h"
+
+namespace ProcessLib
+{
+CoordinateSystem::CoordinateSystem(Parameter<double> const& e0,
+                                   Parameter<double> const& e1)
+    : _base{&e0, &e1, nullptr}
+{
+    if (typeid(_base[0]) != typeid(_base[1]))
+    {
+        OGS_FATAL(
+            "The parameter types for the basis must be equal but they are '%s' "
+            "and '%s'.",
+            typeid(_base[0]).name(),
+            typeid(_base[1]).name());
+    }
+    if (_base[0]->isTimeDependent() || _base[1]->isTimeDependent())
+    {
+        OGS_FATAL("The parameters for the basis must not be time dependent.");
+    }
+    if (_base[0]->getNumberOfComponents() != 2 ||
+        _base[1]->getNumberOfComponents() != 2)
+    {
+        OGS_FATAL("The parameters for the 2D basis must have two components.");
+    }
```

Comment:
In case there are wrong input by mistakes, it is also good to check the values of the bases, such as e_i \dot e_j := (i==j) ? 1, 0.

Reply:
Checking the determinant of the transformation matrix to be close to 1. In debug build only, though.

Reply:
I see.